modules/sdk-coin-ada/test/unit/keyPair.ts

```diff
@@ -0,0 +1,111 @@
+
+      // keyPair = new KeyPair({
+      //   prv: CardanoWasm.Bip32PrivateKey.from_bech32(enterpriseAccounts.account2.secretKey).to_raw_key().to_bech32()
+      // });
+      // should.equal(keyPair.getKeys().prv, enterpriseAccounts.account2.secretKey);
+      // should.equal(keyPair.getKeys().pub, enterpriseAccounts.account2.publicKey);
+
+      // keyPair = new KeyPair({
+      //   prv: CardanoWasm.Bip32PrivateKey.from_bech32(enterpriseAccounts.account3.secretKey).to_raw_key().to_bech32()
+
+      // });
+      // should.equal(keyPair.getAddress('testnet'), enterpriseAccounts.account3.baseAddress);
+      // should.equal(keyPair.getKeys().pub, enterpriseAccounts.account3.publicKey);
+    });
+
+    //   it('initialization from public key', () => {
+    //     let keyPair = new KeyPair({ pub: account1.publicKey });
+    //     should.equal(keyPair.getKeys().pub, account1.publicKey);
+
+    //     keyPair = new KeyPair({ pub: account4.publicKey });
+    //     should.equal(keyPair.getAddress(), account4.address);
+    //   });
+    // });
+
+    describe('KeyPair validation', () => {
+      it('should fail to create from an invalid seed', () => {
+        const seed = { seed: Buffer.alloc(8) }; //  Seed should be 512 bits (64 bytes)
+        should.throws(() => new KeyPair(seed), 'bad seed size');
+      });
+
+      it('should fail to create from an invalid public key', () => {
+        const source = {
+          pub: '01D63D',
+        };
+        should.throws(() => new KeyPair(source), 'address seems to be malformed');
+      });
+
+      it('should fail to create from an invalid private key', () => {
+        const source = {
+          prv: '82A34',
+        };
+        should.throws(() => new KeyPair(source), 'Invalid base32 characters');
+      });
+    });
+
+    // describe('getAddress', () => {
+    //   it('should get an address', () => {
+    //     let keyPair = new KeyPair({ prv: accounts.account1.secretKey });
+    //     let address = keyPair.getAddress();
+    //     address.should.equal(accounts.account1.address);
+
+    //     keyPair = new KeyPair({ prv: accounts.account2.secretKey });
+    //     address = keyPair.getAddress();
+    //     address.should.equal(accounts.account2.address);
+    //   });
+    // });
+  });
+
+  describe('getKeys', () => {
+    it('should get private and public keys in the protocol default format', () => {
+      const keyPair = new KeyPair(defaultSeed);
+      const { prv, pub } = keyPair.getKeys();
+
+      prv!.should.equal(prv);
+      pub.should.equal(pub);
+      // const address = keyPair.getAddress('testnet')
+      // address.should.equal(accounts.seedAccount.enterpriseAddress);
+    });
+
+    it('should get private and public keys for a random seed', () => {
+      const keyPair = new KeyPair();
+      const { prv, pub } = keyPair.getKeys();
+      should.exist(prv);
+      should.exist(pub);
+    });
+  });
```

Comment:
do we have a testcase that create key-pair from public key ( in hex) and verify address?